cli/cli/kurtosis_gateway/connection/provider.go

```diff
@@ -93,10 +93,12 @@ func (provider *GatewayConnectionProvider) ForEnclaveApiContainer(enclaveInfo *k
 	return apiContainerConnection, nil
 }
 
-func (provider *GatewayConnectionProvider) ForUserService(enclaveId string, serviceUuid string, servicePortSpecs map[string]*port_spec.PortSpec) (GatewayConnectionToKurtosis, error) {
-	podPortforwardEndpoint, err := provider.getUserServicePodPortforwardEndpoint(enclaveId, serviceUuid)
+func (provider *GatewayConnectionProvider) ForMaybeUserService(enclaveId string, serviceUuid string, servicePortSpecs map[string]*port_spec.PortSpec) (GatewayConnectionToKurtosis, error) {
```

Comment:
Rename some methods to include `Maybe` to indicate that this might not happen since it does not happen for stopped services.  Added a comment to `writeMaybeOverServiceInfoFieldsWithLocalConnectionInformation` to also clarify things.  We can revert those changes once the service status (registered, stopped, started) is part of the `GetServices` response since we can easily check if we need to call `writeOverServiceInfoFieldsWithLocalConnectionInformation`.